src/api/journalEntries/journalEntry.ts

```diff
@@ -90,12 +90,22 @@ export const createJournalEntry = async (createJournalEntryInput: JournalEntryCr
   };
 
   const { data } = await API.mutate<any>({
-    mutation: gql(createJournalEntryQuery),
+    mutation: createJournalEntryMutation,
     variables: { data: variables}
   });
   
   return data.createJournalEntry;
-};
+}
+
+export const useCreateJournalEntry = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation(createJournalEntry, {
+    onSuccess: () => {
+      queryClient.invalidateQueries("journalEntries")
+    }
```

Comment:
On successful creation:
- the cache invalidates the journalEntries query if there is one 
- if there isn't one it creates one
- if there is one cached then it will be marked as "stale" I believe which triggers react-query to refetch the "journalEntries" query and getting the most updated data